### ISSUE TITLE
empty blocks should be okay in testing

### ISSUE THREAD

**Issue by shriram** [brownplt/pyret-lang#1216]
If I have a test block like
```
check:
  f(10) is 3
end
```
and let's say `f(10)` is causing an infinite loop. When I stop the program, none of my definitions are available, so I can't even try to diagnose what, say, `f(0)` produces. So I comment out this one line and try to run the program again. But now I get an error saying that the block is empty.

While true, this is unnecessarily onerous. It seems entirely legitimate for someone to have an empty `check`, `where`, or `examples` block. These are different kinds of blocks than the bodies of functions, say, which need to have something to produce a value.

**Comment by blerner** [comment]
IIRC, the counterargument you/we/someone had made was, "Why write a testing block if you have no tests?  At least put a ... or something in there to indicate it's incomplete"  This is a different sort of use case than empty-block checking.  In our JS unit test framework (or in ruby's, or in many others), we have a test-runner function called "it" (as in, "it does...") that we can easily comment out by renaming it to "xit" (as in, "X it out").  I'd be happy adding new keywords, `skip-check`, `skip-where`, `skip-examples` that temporarily ignore a block of tests (but still parse and scope-check and such over them, so they don't bitrot).

**Comment by schanzer** [comment]
I would actually _love_ the ability to have an empty `examples:` block. This allows us to start students off with a perfectly runnable file, but clearly indicate where tests should be written. It also reduces the amount of typing (and typos) for our younger users, by having them focus on writing the tests and not the keywords.

**Comment by jpolitz** [comment]
Yeah I agree with this. Empty check/examples blocks seem useful in a way
that other empty blocks aren’t.

**Comment by blerner** [comment]
Should they render in the output in a different color, though?  I don't really think like `check: end` is a _successful_ check block, nor do I think it's a failed one...

**Comment by shriram** [comment]
Yes, absolutely! In terms of printed output, you haven't passed all tests, either, if you've passed zero tests.

**Comment by blerner** [comment]
Well, to be ever so slightly pedantic, you have, in the same sense that andmap over an empty list should return true ;-)

I worry a bit about what color to choose, though, since we already have pink/yellow/green/blue in use... Orange, meaning halfway between a failed test and an erroring test?  Gray, meaning boring?

**Comment by shriram** [comment]
We talked about this a few weeks ago. It was the exchange about a situation where someone writes `==` instead of `is` in a check block, meaning there are only expressions, no actual checks, in that block. 

We said that that block clearly has not failed, but it should not be considered to have passed, either. We said we'd need a third color for that situation, while noting that if someone mixes `==` and `is`, that color doesn't help (unless one extends that color to cover cases where there are any boolean-valued top-level expressions in the block…). 

I don't want to make new policy here; hoping someone can dig that up.

**Comment by blerner** [comment]
See #1231 -- we didn't like empty check blocks there, though... (moreso about the display of them rather than their mere existence)

**Comment by shriram** [comment]
I think these are in harmony. A third color, for empty blocks, and a message saying "there were no checks in this block" (saying nothing about passed or failed) seems about right. I feel like that covers both this case and #1231.
